lld/ELF/Config.h

```diff
@@ -679,6 +681,25 @@ static inline void internalLinkerError(StringRef loc, const Twine &msg) {
               llvm::getBugReportMsg());
 }
 
+struct ELFSyncStream : SyncStream {
+  Ctx &ctx;
+  ELFSyncStream(Ctx &ctx, DiagLevel level)
+      : SyncStream(*ctx.e, level), ctx(ctx) {}
+};
+
+template <typename T>
+std::enable_if_t<!std::is_pointer_v<std::remove_reference_t<T>>,
+                 const ELFSyncStream &>
+operator<<(const ELFSyncStream &s, T &&v) {
+  s.os << std::forward<T>(v);
+  return s;
+}
+
+ELFSyncStream Log(Ctx &ctx);
+ELFSyncStream Warn(Ctx &ctx);
+ELFSyncStream Err(Ctx &ctx);
```

Comment:
Is the intention that all errors are now errorOrWarn, it looks we currently have error and errorOrWarn. 

I can see error being useful for conditions where the output file is always going to be broken, but we can complete the link with diagnostics. The errorOrWarn can be for when the output is likely broken, or some standard has been broken. FatalError can be used if we're going crash if we keep going.

This isn't a strong distinction so I can see it being worthwhile to merge so that more diagnostics can become warnings if needed.

Reply:
Agree with the above. I think we should add a comment before these 4 functions, to explain in which situation they should be used, and what are the expectations.

Reply:
We prefer `errorOrWarn` to `error` and the former is used much more. The lengthy name problem can now be "fixed" as we are switching to `Err`.

I agree that we need a replacement for `lld::error` as well. Do you have a suggestion for the name?

Reply:
If `Err` is the default, then perhaps `ErrAlways` for the non downgradeable variety. If it isn't as common it could have a longer name.

Alternatives: `ForceErr`, `PermErr`, `AlwaysErr`.

Reply:
Adopted `ErrAlways`.

Sorry that I just realized that I did not update this yet.